tests/plugins/test_koji_promote.py

```diff
@@ -267,13 +266,19 @@ def mock_environment(tmpdir, session=None, name=None,
         sync_result = {}
         for image in workflow.tag_conf.images:
             tag = image.to_str(registry=False)
-            digests = ManifestDigest(v1=fake_digest(image, 1),
-                                     v2=fake_digest(image, 2))
-            docker_reg.digests[tag] = digests
+            man_fake = ManifestDigest(v1=fake_digest(image, 1),
+                                      v2=fake_digest(image, 2))
+
+            if digests:
+                man_digests = ManifestDigest(**digests)
+            else:
+                man_digests = man_fake
+
+            docker_reg.digests[tag] = man_digests
             if pulp_registries:
-                sync_result[digests.v1] = {}
+                sync_result[man_fake.v1] = {}
                 if pulp_supports_schema2:
-                    sync_result[digests.v2] = {}
+                    sync_result[man_fake.v2] = {}
```

Comment:
Same comment as above, this looks to me like it should be `sync_result[man_digests.v2] = {}`.

Reply:
same as reply above

Reply:
I think *this* line should be `man_digests.v2` (with improved variable naming), because the v2 schema 2 manifest digest in the registry will always match what gets synced to Pulp.

Reply:
ok will do